keylime/agentstates.py

```diff
@@ -1,9 +1,47 @@
+import sys
 import threading
 
 from keylime.common.algorithms import Hash
 from keylime.ima.ast import get_FF_HASH, get_START_HASH
 from keylime.ima.file_signatures import ImaKeyrings
 
+if sys.version_info >= (3, 7):
+    from dataclasses import dataclass
+else:
+    from keylime.backport_dataclasses import dataclass
+
+
+@dataclass
+class TPMClockInfo:
+    clock: int
+    resetcount: int
+    restartcount: int
+    safe: int
+
+    @classmethod
+    def from_json(cls, data) -> "TPMClockInfo":
+        dclki = {}
+        if "clockInfo" in data:
+            dclki = data["clockInfo"]
+
+        if "clock" in data:
+            dclki = data
+
+        return cls(
+            clock=dclki.get("clock", 0),
+            resetcount=dclki.get("resetCount", 0),
+            restartcount=dclki.get("restartCount", 0),
+            safe=dclki.get("safe", 1),
+        )
+
+    def to_json(self):
```

Comment:
`to_json()` is probably a bit confusing because we are not actually converting to json. It produces a `dict`.